### PR TITLE
[risk=low][no ticket] rm all calls to getCompletionTime() and getBypassTime()

### PR DESCRIPTION
**Pull request by jmthibault79**
Use accessModuleService as the source of truth in all cases

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java [hunk 1]
```diff
@@ -841,7 +841,13 @@ public DbUser syncTwoFactorAuthStatus(DbUser targetUser, Agent agent, boolean is
     return updateUserWithRetries(
         user -> {
           if (isEnrolledIn2FA) {
-            if (user.getTwoFactorAuthCompletionTime() == null) {
+            final boolean hasTwoFactorCompletion =
+                accessModuleService
+                    .getAccessModuleStatus(user, AccessModuleName.TWO_FACTOR_AUTH)
+                    .map(status -> status.getCompletionEpochMillis() != null)
+                    .orElse(false);
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java:848]: if the user has no status for this module, obviously it's not complete

**Review by yonghaoy — APPROVED** [review]

**Review by calbach — APPROVED** [review]
How is this not currently causing issues? Is it because we're still dual writing these columns?

**Comment by jmthibault79** [comment]
yes

File: api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java [hunk 2]
```diff
@@ -161,9 +161,8 @@ public boolean isModuleBypassed(DbUser dbUser, AccessModuleName accessModuleName
    */
   private DbUserAccessModule retrieveUserAccessModuleOrCreate(
       DbUser user, DbAccessModule dbAccessModule) {
-    return userAccessModuleDao.getAllByUser(user).stream()
-        .filter(m -> m.getAccessModule().getName().equals(dbAccessModule.getName()))
-        .findFirst()
+    return userAccessModuleDao
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java:164]: Should be more efficient

File: api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java [hunk 3]
```diff
@@ -141,6 +141,8 @@ public void setUp() {
 
     providedWorkbenchConfig = WorkbenchConfig.createEmptyConfig();
     providedWorkbenchConfig.accessRenewal.expiryDays = (long) 365;
+    providedWorkbenchConfig.access.enableComplianceTraining = true;
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/db/dao/UserServiceTest.java:144]: new-style is more sensitive to feature flags, so tests were failing before this change

**Pull request merged**